crates/networking/p2p/src/network.rs

```diff
@@ -406,9 +517,102 @@ mod tests {
         Network::init(executor, &config).await
     }
 
+    #[test]
+    fn peer_id_from_enr_matches_libp2p() {
+        let lp_kp = Keypair::generate_secp256k1();
+        let secret = lp_kp
+            .clone()
+            .try_into_secp256k1()
+            .unwrap()
+            .secret()
+            .to_bytes();
+        let signing = SigningKey::from_slice(&secret).unwrap();
+
+        let enr_key = CombinedKey::Secp256k1(signing);
+        let enr = Enr::builder().build(&enr_key).unwrap();
+
+        let expected = PeerId::from_public_key(&lp_kp.public());
+        let actual = Network::peer_id_from_enr(&enr).expect("peer id");
+
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn insert_then_read_returns_snapshot() {
+        init_network_spec();
+
+        let rt = tokio::runtime::Runtime::new().unwrap();
+
+        let mut net = rt.block_on(async {
+            create_network("127.0.0.1".parse().unwrap(), 0, 0, vec![], true, vec![])
+                .await
+                .unwrap()
+        });
+
+        let peer_id = libp2p::PeerId::random();
```

Comment:
```suggestion
        let peer_id = PeerId::random();
```

everywhere else as well